### ISSUE TITLE
application.name field became mandatory/redundant in release 0.7.x

### ISSUE THREAD

**Issue by nastacio** [redhat-developer/service-binding-operator#965]
### What is the environment (Minikube, Openshift)?
OCP 4.6.16 on premises

### What is the SBO version used?
0.7.0
0.7.1

### What are the steps to reproduce this issue?
  1. Install the SBO operator (see attached `subscription.yaml` in attached archive)
  2. Deploy a sample app, using [these instructions](https://github.com/redhat-developer/service-binding-operator/tree/master/examples/nodejs_postgresql#import-an-application)
  3. Apply the binding request (see `binding-request-noname.yaml` in attached archive) , which uses label selectors to match the application:
      ```
      oc project service-binding-demo
      oc apply -f binding-request-noname.yaml
      ```
  4. Inspect logs for SBO pod (see `logs` in attached archive)
      ```
     {"level":"error","ts":1621375914.120951,"logger":"controller","msg":"Reconciler error","reconcilerGroup":"binding.operators.coreos.com","reconcilerKind":"ServiceBinding","controller":"servicebinding","name":"binding-request-noname","namespace":"service-binding-demo","error":"name is required"
      ```
  5. Now "fix" the `ServiceBinding` by adding the redundant `name` field:
      ```
      oc apply -f binding-request-with-name.yaml
      ```
  6. Inspect logs for SBO pod (see `logs` in attached archive) and see a successful message:
      ```
    {"level":"info","ts":1621376188.5513623,
     "logger":"controllers.ServiceBinding",
     "msg":"Done","serviceBinding":"service-binding-demo/binding-request-name"
      ```

### What is the expected behaviour?
Service binding should be processed without issue and mapped to the matching application.

### What is the actual behaviour?
Operator logs surface an error in the binding request, indicating that the `.spec.application.name` must be filled out, even though `.spec.application.labelSelector` is specified.

### Service Binding Operator Logs
See `logs` in attached archive.

### Additional Information (Screenshots, etc)
This same `binding-request-noname.yaml` file worked correctly ysing the operator version 0.6.0.
I can see the secret created by the operator:

```
 oc get Secret binding-request-noname-6f93c924eb98fbd528c74b610b632229352734c1 -n service-binding-demo -o jsonpath={.data}
map[env.json:ew...==]
```

And I can see it mounted to the application
```
oc get pod nodejs-rest-http-crud-7dfd6855db-r5s5s -n service-binding-demo -o jsonpath="{.spec.containers[].volumeMounts}" 
[map[name:binding-request-noname mountPath:/config] ...
```

[Archive.zip](https://github.com/redhat-developer/service-binding-operator/files/6504896/Archive.zip)

**Comment by pedjak** [comment]
I can confirm this regression, we will fix it as a part of 0.8 release.

**Comment by baijum** [comment]
Jira issue: https://issues.redhat.com/browse/APPSVC-883
